vault/resource_jwt_auth_backend.go

```diff
@@ -263,7 +307,24 @@ func jwtAuthBackendRead(d *schema.ResourceData, meta interface{}) error {
 		if configOption == "oidc_client_secret" {
 			continue
 		}
-		d.Set(configOption, config.Data[configOption])
+
+		if configOption == "provider_config" {
+			var keys []string
+			rawProviderConfig := config.Data[configOption].(map[string]interface{})
+			for k := range rawProviderConfig {
+				keys = append(keys, k)
+			}
+
+			sort.Sort(sort.StringSlice(keys))
```

Comment:
I think this could be sort.Strings():
```suggestion
			sort.Strings(keys)
```

Though I'm curious why the provider_config keys need to be sorted. Does it have something to do with HasChanged()?

Reply:
Yeah, the sort is required for the hash computed by Terraform. I'll add a comment about this.

Reply:
Sorting worked and I added a comment.